Frameworks/AEPSwiftUI/Sources/ContentCards/ContentCardUIEventListening.swift

```diff
@@ -0,0 +1,33 @@
+/*
+ Copyright 2024 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import Foundation
+
+public protocol ContentCardUIEventListening {
```

Comment:
wondering if we should provide default implementations for these (basically to make them optional for the app developer implementing this protocol).

thoughts?

Reply:
Implementation of this protocol is still optional for the developer
```
 public static func getContentCardsUI(for surface: Surface,
                                         customizer: ContentCardCustomizing? = nil,
                                         listener: ContentCardUIEventListening? = nil,
                                         _ completion: @escaping (Result<[ContentCardUI], Error>
```

the API can be called without passing in the listener
```
 AEPSwiftUI.getContentCardsUI(for: pdfPageSurface,
                                     customizer: PDFPageCardCustomizer()) { result in
                    // handle result 
        }
```



If the developer does not provide